src/main/java/org/codeforamerica/messaging/providers/mailgun/MailgunCallbackController.java

```diff
@@ -35,27 +35,33 @@ public MailgunCallbackController(EmailMessageRepository emailMessageRepository,
 
     @PostMapping(path = "/status")
     public ResponseEntity<Object> updateStatus(@RequestBody JsonNode requestJSON) {
+        String providerMessageId = requestJSON.at("/event-data/message/headers/message-id").textValue();
+
         if (!mailgunSignatureVerificationService.verifySignature(requestJSON)) {
-            log.error("Signature verification failed");
+            log.error("Signature verification failed. Provider message id: {}", providerMessageId);
             return new ResponseEntity<>("Unauthorized", HttpStatus.UNAUTHORIZED);
         }
 
-        EmailMessage emailMessage = emailMessageRepository.findFirstByProviderMessageId(
-                requestJSON.at("/event-data/message/headers/message-id").textValue());
+        EmailMessage emailMessage = emailMessageRepository.findFirstByProviderMessageId(providerMessageId);
         String rawEmailStatus = requestJSON.at("/event-data/event").textValue();
         if (rawEmailStatus.equals("unsubscribed")) {
             unsubscribeEmail(requestJSON);
             return ResponseEntity.ok().build();
         }
-        MessageStatus status = mapMailgunStatustoMessageStatus(rawEmailStatus);
-        emailMessage.getMessage().setRawEmailStatus(rawEmailStatus);
-        emailMessage.getMessage().setEmailStatus(status);
-        if (hadError(status)) {
-            emailMessage.setProviderError(buildProviderError(requestJSON, status));
+        MessageStatus newEmailStatus = mapMailgunStatustoMessageStatus(rawEmailStatus);
+        MessageStatus oldEmailStatus = emailMessage.getMessage().getEmailStatus();
+        if (oldEmailStatus == null || newEmailStatus.compareTo(oldEmailStatus) > 0) {
```

Comment:
Is there any way you can think of to make `> 0` not a magic number? It's relying on ranking of the enum values, right? I'm ok leaving it, just wondering if you have other thoughts on how to improve readability.

Reply:
0 is not all that magic here because `Comparable.compareTo` has a well-known behavior. However, I have taken your suggestion regarding readability and moved it to a better named method (`isAfter`) in `MessageStatus` itself.